rotkehlchen/data_import/importers/bitmex.py

```diff
@@ -0,0 +1,166 @@
+
+        margin_position = MarginPosition(
+            location=Location.BITMEX,
+            open_time=None,
+            close_time=close_time,
+            profit_loss=realised_pnl,
+            pl_currency=asset_balance.asset,
+            fee=fee,
+            fee_currency=asset_balance.asset,
+            notes=notes,
+            link=f'Imported from BitMEX CSV file. Transact Type: {csv_row["transactType"]}',
+        )
+        history_event = HistoryEvent(
+            event_identifier=event_identifier,
+            sequence_index=0,
+            timestamp=ts_sec_to_ms(close_time),
+            location=Location.BITMEX,
+            asset=asset_balance.asset,
+            balance=asset_balance.balance,
+            notes=notes,
+            event_type=HistoryEventType.MARGIN,
+            event_subtype=history_event_subtype,
+        )
+
+        return margin_position, history_event
+
+    @staticmethod
+    def _consume_deposits_or_withdrawals(
+            csv_row: dict[str, Any], timestamp_format: str = '%m/%d/%Y, %H:%M:%S %p',
+    ) -> AssetMovement:
+        asset = A_BTC.resolve_to_asset_with_oracles()
+        amount = deserialize_asset_amount_force_positive(csv_row['amount'])
+        fee = deserialize_fee(csv_row['fee'])
+        transact_type = csv_row['transactType']
+        category = AssetMovementCategory.DEPOSIT if transact_type == 'Deposit' else AssetMovementCategory.WITHDRAWAL  # noqa: E501
+        amount = AssetAmount(satoshis_to_btc(amount))  # bitmex stores amounts in satoshis
+        fee = Fee(satoshis_to_btc(fee))
+        ts = deserialize_timestamp_from_date(
+            date=csv_row['transactTime'],
+            formatstr=timestamp_format,
+            location='Bitmex Wallet History Import',
+        )
+        return AssetMovement(
+            location=Location.BITMEX,
+            category=category,
+            address=deserialize_asset_movement_address(csv_row, 'address', asset),
+            transaction_id=get_key_if_has_val(csv_row, 'tx'),
+            timestamp=ts,
+            asset=asset,
+            amount=amount,
+            fee_asset=asset,
+            fee=fee,
+            link=f'Imported from BitMEX CSV file. Transact Type: {transact_type}',
+        )
+
+    def _import_csv(self, cursor: DBCursor, filepath: Path, **kwargs: Any) -> None:
+        """
+        Import deposits, withdrawals and realised pnl events from BitMEX.
+        May raise:
+        - UnsupportedCSVEntry if operation not supported
+        - InputError if a column we need is missing
+        """
+        history_events = []
+        with open(filepath, encoding='utf-8-sig') as csvfile:
+            data = csv.DictReader(csvfile)
+            for row in data:
+                try:
+                    if row['transactType'] == 'RealisedPNL':
+                        margin_position, history_event = self._consume_realised_pnl(row, **kwargs)
+                        self.add_margin_trade(cursor, margin_position)
```

Comment:
So as I wrote in discord this is something that can't work. You should only keep the `MarginPosition` here. As we are duplicating the event otherwise.

We have open issues to break down everything into `HistoryEvent` eventually.

https://github.com/rotki/rotki/issues/6097
https://github.com/rotki/rotki/issues/6096

And possibly we need more. We also need for Margin positions. So when these would be completed we would duplicate any imported info.

So for now if you want to import margin positions, import them simply as MarginPosition.

You could in a next step:

1. Create an issue to turn `MarginPosition` into HistoryEvent.
2. Then implement it by:
    a. Removing MarginPosition
    b. Adding a user DB upgrade to turn all existing MarginPosition to HistoryEvent
    c. Replace all usages of MarginPosition with `HistoryEvent`
    d. Make sure accounting/PnL works fine for them